src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPointParser.java

```diff
@@ -71,101 +78,158 @@ public static XYPoint parseXYPoint(Object value, final boolean ignoreZValue) thr
-            try (XContentSubParser subParser = new XContentSubParser(parser)) {
-                while (subParser.nextToken() != XContentParser.Token.END_OBJECT) {
-                    if (subParser.currentToken() != XContentParser.Token.FIELD_NAME) {
-                        throw new OpenSearchParseException("token [{}] not allowed", subParser.currentToken());
-                    }
-                    String field = subParser.currentName();
-                    if (!(X_PARAMETER.equals(field) || Y_PARAMETER.equals(field))) {
-                        throw new OpenSearchParseException("field must be either [{}] or [{}]", X_PARAMETER, Y_PARAMETER);
-                    }
-                    if (X_PARAMETER.equals(field)) {
-                        subParser.nextToken();
-                        switch (subParser.currentToken()) {
-                            case VALUE_NUMBER:
-                            case VALUE_STRING:
-                                try {
-                                    x = subParser.doubleValue(TRUE);
-                                } catch (NumberFormatException numberFormatException) {
-                                    throw new OpenSearchParseException("[x] must be valid double value", numberFormatException);
-                                }
-                                break;
-                            default:
-                                throw new OpenSearchParseException("[x] must be a number");
-                        }
-                    }
-                    if (Y_PARAMETER.equals(field)) {
-                        subParser.nextToken();
-                        switch (subParser.currentToken()) {
-                            case VALUE_NUMBER:
-                            case VALUE_STRING:
-                                try {
-                                    y = subParser.doubleValue(TRUE);
-                                } catch (NumberFormatException numberFormatException) {
-                                    throw new OpenSearchParseException("[y] must be valid double value", numberFormatException);
-                                }
-                                break;
-                            default:
-                                throw new OpenSearchParseException("[y] must be a number");
-                        }
-                    }
-                }
+        switch (parser.currentToken()) {
+            case START_OBJECT:
+                parseXYPointObject(parser, point, ignoreZValue);
+                break;
+            case START_ARRAY:
+                parseXYPointArray(parser, point, ignoreZValue);
+                break;
+            case VALUE_STRING:
+                String val = parser.text();
+                point.resetFromString(val, ignoreZValue);
+                break;
+            default:
+                throw new OpenSearchParseException("xy_point expected");
```

Comment:
we should enhance this error message to something like expecting xy_point as an array, string or object format.

Reply:
ack